RIOT-OS/RIOT
# cpu/cortexm_common: Add support for Cortex-M interrupt sub-priorities

### Contribution description

This commit enables Cortex-M CPU interrupt sub-priorities and allows the PendSV interrupt to have a priority different from the default one.  Together these two new preprocessor defines can be used to have PendSV always run as the last interrupt, before returning from the interrupt stack back to the user space.

Running PendSV as the last interrupt before returning to the user space is recommended by ARM, as it increases efficiency.  Furthermore, that change enhances stability a lot with the new nRF52 SoftDevice support, currently being worked in PR #9473.

This commit merely enables sub-priorities and a separate PendSV priority to be used without changing the default RIOT behaviour.

### Testing procedure

As such, this should not change anything.  All the previous tests should run as before, and no new ones are introduced.

### Issues/PRs references

See also #9473 and https://github.com/RIOT-OS/RIOT/issues/10122
